src/components/Form/Form.tsx

```diff
@@ -23,6 +23,7 @@ type Props<T extends FormDetails> = {
   onDraftRestore?: (newState: FormState<T>) => void;
   children: (props: FormContextValue<T>) => React.ReactNode;
   hideRestoreDraft?: boolean;
+  showSaveAndAddMoreBtn?: string;
```

Comment:
the prefix `show` makes it read as expected type to be a boolean, but string is expected. This could be confusing.

How about a more generic solution that'd allow us to freely add any type of button and make the actual component that uses this component define what type of button to show specifically?

```suggestion
  additionalButtons?: { type: "submit" | "button"; label: string; id: string }[];
```